components/sharing-dialog/src/helpers/helpers.js

```diff
@@ -0,0 +1,249 @@
+import {
+    ACCESS_VIEW_ONLY,
+    ACCESS_NONE,
+    ACCESS_VIEW_AND_EDIT,
+    SHARE_TARGET_EXTERNAL,
+    SHARE_TARGET_PUBLIC,
+} from '../sharing-constants.js'
+
+/**
+ * Returns a function, that, as long as it continues to be invoked, will not be triggered. The
+ * function will be called after it stops being called for N milliseconds. If `immediate` is
+ * passed, trigger the function on the leading edge, instead of the trailing.
+ */
+
+export const debounce = (func, wait, immediate) => {
+    let timeout
+
+    return (...args) => {
+        const context = this
+
+        const later = () => {
+            timeout = null
+
+            if (!immediate) {
+                func.apply(context, args)
+            }
+        }
+
+        const callNow = immediate && !timeout
+
+        clearTimeout(timeout)
+        timeout = setTimeout(later, wait)
+
+        if (callNow) {
+            func.apply(context, args)
+        }
+    }
+}
+
+/**
+ * Name conversion
+ */
+
+export const nameToInitials = (name) => {
+    if (!name) {
+        return ''
+    }
+
+    const parts = name.split(' ')
+    const firstNameInitial = parts.length > 0 ? parts.shift().charAt(0) : ''
+    const lastNameInitial = parts.length > 0 ? parts.pop().charAt(0) : ''
+
+    return `${firstNameInitial}${lastNameInitial}`
+}
+
+/**
+ * Access and constant conversion
+ */
+
+export const convertAccessToConstant = (access) => {
+    if (typeof access === 'boolean') {
+        return access ? ACCESS_VIEW_ONLY : ACCESS_NONE
+    }
+
+    if (/^rw/.test(access)) {
+        return ACCESS_VIEW_AND_EDIT
+    } else if (/^r-/.test(access)) {
+        return ACCESS_VIEW_ONLY
+    } else {
+        return ACCESS_NONE
+    }
+}
+
+export const convertConstantToAccess = (constant, useBoolean) => {
+    switch (constant) {
+        case ACCESS_NONE:
+            return useBoolean ? false : '--------'
+        case ACCESS_VIEW_ONLY:
+            return useBoolean ? true : 'r-------'
+        case ACCESS_VIEW_AND_EDIT:
+            return useBoolean ? true : 'rw------'
+        default:
+            return useBoolean ? false : '--------'
+    }
+}
+
+/**
+ * Mapping users and groups to style used internally
+ */
+
+export const mapUsers = ({ id, name, access }) => ({
```

Comment:
Function name does not describe what happens, it actually even deceives a bit as I'd expect to pass it an array (applies to `mapGroups` as well

Reply:
maybe `convertUserAccessToConstant`?

Reply:
Yeah this bit didn't have to be as complex, I've replaced it with a function that just picks out the access property here: faaee59cba77a36abae4ad767ce2af4b5408de15